main.py

```diff
@@ -81,12 +79,16 @@ async def create_compilation_task(
 
     # add request contracts in temp directory
     for file in files:
-        content = (await file.read()).decode("utf-8")
-        (project_root / "contracts" / file.filename).write_text(content)
+        content = await file.read()
+        (project_root / "contracts" / file.filename).write_bytes(content)
+
+    config_override = dict()
+    if vyper_version:
+        config_override["vyper"] = dict(version=vyper_version)
```

Comment:
Wonder if we can go a bit higher level and have the config override by a part of the POST method params